pkg/parser/parser.go

```diff
@@ -212,6 +212,37 @@ func (e *expr) Metrics() []MetricRequest {
 					r[i].From = fromNew
 				}
 			}
+		case "hitcount":
+			if len(e.args) < 2 {
+				return nil
+			}
+
+			if len(e.args) == 3 {
+				alignToInterval, err := e.GetBoolNamedOrPosArgDefault("alignToInterval", 2, false)
+				if err != nil {
+					return nil
+				}
+				if alignToInterval {
+					bucketSizeInt32, err := e.GetIntervalArg(1, 1)
+					if err != nil {
+						return nil
+					}
+					interval := int64(bucketSizeInt32)
+					// This is done in order to replicate the behavior in Graphite web when alignToInterval is set,
+					// in which new data is fetched with the adjusted start time.
+					for i, _ := range r {
+						start := from
```

Comment:
I think this needs to take the `From` from the current request; otherwise the offsets from nested functions would be overwritten.  Similar to what is done with the moving functions: https://github.com/go-graphite/carbonapi/pull/759/files#diff-2a6035c752559ce0fb74ced451e9e4363c01598c669d8f77a46be9450e6f6d87L211
```suggestion
						start := r[i].From
```